libs/api/domains/license-service/src/lib/client/firearm-license-client/firearmLicenseService.api.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  GenericLicenseClient,
+  GenericLicenseUserdataExternal,
+  GenericUserLicensePkPassStatus,
+  GenericUserLicenseStatus,
+  PkPassVerification,
+} from '../../licenceService.type'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import {
+  FirearmApplicationApi,
+  LicenseInfo,
+} from '@island.is/clients/firearm-license'
+import { User } from '@island.is/auth-nest-tools'
+import { Inject } from '@nestjs/common'
+import { parseFirearmLicensePayload } from './firearmLicenseMapper'
+
+/** Category to attach each log message to */
+const LOG_CATEGORY = 'firearmlicense-service'
+
+export class GenericFirearmLicenseApi
+  implements GenericLicenseClient<LicenseInfo> {
+  constructor(
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    private firearmApi: FirearmApplicationApi,
+  ) {}
+
+  async fetchLicense(user: User) {
+    let license: unknown
+
+    try {
+      license = await this.firearmApi.apiFirearmApplicationLicenseInfoSsnGet({
+        ssn: user.nationalId,
+      })
+    } catch (e) {
+      this.logger.error('Firearm license fetch failed', {
```

Comment:
Refactor error logging to what was done in _adrLicenseService.api.ts_ to handle 403 and 404 errors differently. Log as info instead of error to minimize error noise in Datadog.